### PR TITLE
POC: histogram widget

### PR DESCRIPTION
**Pull request by Haroenv**
PSA: I haven't yet cleaned up the code & typings, but that can be done after this demo's features have even approved

Try the story: https://deploy-preview-4358--instantsearchjs.netlify.com/stories/?path=/story/refinements-rangeslider--with-histogram

### CODE REVIEW

File: .storybook/decorators/withHits.ts [hunk 1]
```diff
@@ -26,7 +37,14 @@ export const withHits = (
   const urlLogger = action('Routing state');
   const search = instantsearch({
     indexName,
-    searchClient: algoliasearch(appId, apiKey),
+    searchClient: algoliasearch(appId, apiKey, {
+      hosts: [
+        {
+          url: appId.toLowerCase() + '-3.algolia.net',
```
Comment (Haroenv) [inline .storybook/decorators/withHits.ts]: this is because we've only deployed this on c1-test-3

File: src/components/Histogram/Histogram.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      return <Bar items={items} />;
+    }
+    case 'line': {
+      return <Line items={items} />;
+    }
+    case 'area': {
+      return <Area items={items} />;
+    }
+    case 'pathLine': {
+      return <PathLine items={items} />;
```
Comment (Haroenv) [inline src/components/Histogram/Histogram.tsx]: these options are mostly just for the demo

File: src/connectors/histogram/connectHistogram.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          true
+        );
+      },
+
+      render({ instantSearchInstance, results }) {
+        const items = (
+          results._rawResults[0]?.facetDistribution?.[attribute] ?? []
+        )
+          // TODO: width -> count
+          .map(({ start, end, width }) => ({ start, end, count: width }));
```
Comment (Haroenv) [inline src/connectors/histogram/connectHistogram.ts]: we should make width into count @Turanic

File: src/connectors/histogram/connectHistogram.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+          [...current]
+        );
+
+        return nextParams;
+      },
+    };
+  };
+};
+
+function getFallbackFacetDistribution({ buckets, min, max }) {
```
Comment (Haroenv) [inline src/connectors/histogram/connectHistogram.ts]: this can be deleted now

File: src/widgets/histogram/defaultTemplates.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default {
+  item: '{{ count }}',
```
Comment (Haroenv) [inline src/widgets/histogram/defaultTemplates.ts]: this isn't used yet

**Review by devdreamsolution — APPROVED** [review]

**Comment by Haroenv** [comment]
Will revisit this if the engine ever has bandwidth to ship it :)

**Pull request closed without merging**